besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java

```diff
@@ -0,0 +1,303 @@
+import static org.hyperledger.besu.ethereum.core.MiningParameters.Unstable.DEFAULT_POW_JOB_TTL;
+import static org.hyperledger.besu.ethereum.core.MiningParameters.Unstable.DEFAULT_REMOTE_SEALERS_LIMIT;
+import static org.hyperledger.besu.ethereum.core.MiningParameters.Unstable.DEFAULT_REMOTE_SEALERS_TTL;
+import static org.hyperledger.besu.ethereum.core.MiningParameters.UpdatableInitValues.DEFAULT_EXTRA_DATA;
+import static org.hyperledger.besu.ethereum.core.MiningParameters.UpdatableInitValues.DEFAULT_MIN_BLOCK_OCCUPANCY_RATIO;
+import static org.hyperledger.besu.ethereum.core.MiningParameters.UpdatableInitValues.DEFAULT_MIN_TRANSACTION_GAS_PRICE;
+
+import org.hyperledger.besu.cli.util.CommandLineUtils;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.core.ImmutableMiningParameters;
+import org.hyperledger.besu.ethereum.core.ImmutableMiningParameters.UpdatableInitValues;
+import org.hyperledger.besu.ethereum.core.MiningParameters;
+
+import java.util.List;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.slf4j.Logger;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+
+/** The Mining CLI options. */
+public class MiningOptions implements CLIOptions<MiningParameters> {
+
+  @Option(
+      names = {"--miner-enabled"},
+      description = "Set if node will perform mining (default: ${DEFAULT-VALUE})")
+  private Boolean isMiningEnabled = false;
+
+  @Option(
+      names = {"--miner-stratum-enabled"},
+      description = "Set if node will perform Stratum mining (default: ${DEFAULT-VALUE})")
+  private Boolean iStratumMiningEnabled = false;
+
+  @Option(
+      names = {"--miner-stratum-host"},
+      description = "Host for Stratum network mining service (default: ${DEFAULT-VALUE})")
+  private String stratumNetworkInterface = "0.0.0.0";
+
+  @Option(
+      names = {"--miner-stratum-port"},
+      description = "Stratum port binding (default: ${DEFAULT-VALUE})")
+  private Integer stratumPort = 8008;
+
+  @Option(
+      names = {"--miner-coinbase"},
+      description =
+          "Account to which mining rewards are paid. You must specify a valid coinbase if "
+              + "mining is enabled using --miner-enabled option",
+      arity = "1")
+  private Address coinbase = null;
+
+  @Option(
+      names = {"--miner-extra-data"},
+      description =
+          "A hex string representing the (32) bytes to be included in the extra data "
+              + "field of a mined block (default: ${DEFAULT-VALUE})",
+      arity = "1")
+  private Bytes extraData = DEFAULT_EXTRA_DATA;
+
+  @Option(
+      names = {"--min-block-occupancy-ratio"},
+      description = "Minimum occupancy ratio for a mined block (default: ${DEFAULT-VALUE})",
+      arity = "1")
+  private Double minBlockOccupancyRatio = DEFAULT_MIN_BLOCK_OCCUPANCY_RATIO;
+
+  @Option(
+      names = {"--min-gas-price"},
```

Comment:
This option is used when selecting txs for block creation, for any kind of network, so it make sense to group here, and I agree that the `miner` naming is now misleading, since we have different ways of building and proposing blocks, so a renaming could help

Reply:
Agree changing Miner Options to something like "Block Builder Options" makes sense